### PR TITLE
[Hold Web-E 30480] Added chronosInCash beta

### PR DESCRIPTION
**Pull request by deetergp**
PullerBear chooses @thienlnam 

Held on https://github.com/Expensify/Web-Expensify/pull/30480. I could probably just pull the beta bits out of that PR, and just commit them on their own.

cc @flodnv @marcaaron 

### Details
Added a beta to Web and E.cash to prevent non-beta users from being able to chat with Chronos.

### Related Issues
* https://github.com/Expensify/Expensify/issues/157087

### Tests
1. Fliped the boolean [on this line](https://github.com/Expensify/Web-Expensify/blob/1e6c6da80383e543f15944726aa35ab4cf01cbe0/lib/BetaManager.php#L87) to `false`.
2. Used Betatify to add the `chronosInCash` beta to my scott.hasbeta@deetfoo.com
	* Betatify seemed broken locally, so I added the beta [here](https://github.com/Expensify/Web-Expensify/blob/6072d58bd02d357eaa189e2656c3f45e2049a254/lib/BetaManager.php#L218) and it worked fine
3. Logged in as scott.hasbeta@deetfoo.com
4. Typed `chronos@expensify.com` and verified that the user appeared as someone I could start a chat with.
	Screen Shot 2021-04-07 at 4 19 23 PM
5. Logged out
6. Logged in as scott.hasnobeta@deetfoo (stashing the change to line 218)
7. Performed step 4 and verified that I did not have the option to begin a chat with Chronos
	Screen Shot 2021-04-07 at 4 20 17 PM

I performed these same steps on all platforms.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Screen Shot 2021-04-07 at 5 27 04 PM
Screen Shot 2021-04-07 at 5 26 42 PM

#### Desktop

Screen Shot 2021-04-07 at 4 19 23 PM
Screen Shot 2021-04-07 at 4 20 17 PM

#### iOS

Screen Shot 2021-04-07 at 4 29 37 PM
Screen Shot 2021-04-07 at 4 23 31 PM

#### Android

Screen Shot 2021-04-07 at 4 38 12 PM
Screen Shot 2021-04-07 at 4 40 53 PM

### CODE REVIEW

**Comment by deetergp** [comment]
Removing the request for @MariaHCD to review since this is a WIP that I won't get back to till Monday.

**Comment by deetergp** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by deetergp** [comment]
In order to test this, I need to apply this diff:
```git
diff --git a/lib/BetaManager.php b/lib/BetaManager.php
index 1395d91b563..6ee283adcd5 100644
--- a/lib/BetaManager.php
+++ b/lib/BetaManager.php
@@ -83,7 +83,7 @@ class BetaManager
      *
      * @var bool
      */
-    private static $allBetasEnabled = true;
+    private static $allBetasEnabled = false;

     /**
      * This property only is used while on DEBUG mode, it indicates if a beta is enabled or not. Mostly for testing.
```

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -300,6 +306,15 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
         userToInvite.icons = [defaultAvatarForUserToInvite];
     }
 
+    if (!isInChronosBeta && searchValue === 'chronos@expensify.com') {
```
Comment (marcaaron) [inline src/libs/OptionsListUtils.js:309]: Should we create a constant for `chronos@expensify.com`?

File: src/libs/OptionsListUtils.js [hunk 2]
```diff
@@ -300,6 +306,15 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
...
+        userToInvite = null;
+        return {
+            personalDetails: [],
+            recentReports: [],
+            userToInvite,
+        };
```
Comment (marcaaron) [inline src/libs/OptionsListUtils.js:315]: Hmm wait why we care if they are even in the beta or not? Simple to just add another condition above like

```js
&& (searchValue !== 'chronos@expensify.com')
```

then just don't set the `userToInvite` if they are trying to find this guy?

**Review by marcaaron — COMMENTED** [review]
LGTM but had a couple of comments. 

Do we also want to avoid having `chronos@expensify.com` show up as a user in the `personalDetails` array returned from `getOptions()`?

File: src/libs/OptionsListUtils.js (hunk 2 above)
Comment (marcaaron) [reply src/libs/OptionsListUtils.js:315]: Sorry nvm ignore me I didn't read any of the description lol

Comment (marcaaron) [reply src/libs/OptionsListUtils.js:315]: Ok well something like this in there somewhere would work

```js
|| (isInChronosBeta && searchValue === 'chronos@expensify.com')
```

Comment (deetergp) [reply src/libs/OptionsListUtils.js:315]: `||` didn't work but `&& (isInChronosBeta && searchValue === CONST.EMAIL.CHRONOS)` did.

Comment (deetergp) [reply src/libs/OptionsListUtils.js]: Okay, what ended up working was adding `&& (searchValue !== CONST.EMAIL.CHRONOS || isInChronosBeta)` to the end if that lengthy condition check.

**Comment by deetergp** [comment]
Updated!

File: src/libs/OptionsListUtils.js (hunk 2 above)
Comment (marcaaron) [reply src/libs/OptionsListUtils.js]: Nice! yeah I knew some combination would 😝

**Review by marcaaron — DISMISSED** [review]
LGTM

**Review by thienlnam — DISMISSED** [review]

**Comment by deetergp** [comment]
Updated to resolve some merge conflicts and some `master` vs. `main` confusion.

**Review by thienlnam — APPROVED** [review]

**Comment by thienlnam** [comment]
This can be merged once the pending PR is on production right?

**Comment by deetergp** [comment]
I don't think anything bad would happen if we merged this now, but I'm not 100% sure about it, so we might as well play it safe and wait.

**Pull request merged**